src/module_writer.rs

```diff
@@ -239,7 +240,26 @@ pub struct WheelWriter {
     wheel_path: PathBuf,
     file_tracker: FileTracker,
     excludes: Override,
-    compression_level: u16,
+    compression: CompressionOptions,
+}
+impl CompressionOptions {
+    fn get_file_options(&self) -> zip::write::FileOptions<()> {
+        let compression_level = self.compression_level;
+        // Unlike users which can use the develop subcommand, the tests have to go through
+        // packing a zip which pip than has to unpack. This makes this 2-3 times faster
+        let compression_method = if cfg!(feature = "faster-tests") || compression_level == 0 {
+            zip::CompressionMethod::Stored
+        } else {
+            self.compression_method.into()
+        };
+
+        let mut options =
+            zip::write::SimpleFileOptions::default().compression_method(compression_method);
+        if compression_level != 0 {
+            options = options.compression_level(Some(compression_level as i64));
```

Comment:
I'm a bit worried that the compression level value varies among compression methods: https://docs.rs/zip/latest/zip/write/struct.FileOptions.html#method.compression_level

see also https://github.com/PyO3/maturin/discussions/2608 which could be related.